cylc/flow/config_schema.py

```diff
@@ -0,0 +1,609 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define all legal items and values for cylc suite definition files."""
+
+import os
+import re
+
+from metomi.isodatetime.data import Calendar
+
+from cylc.flow import LOG
+from cylc.flow import __version__ as CYLC_VERSION
+from cylc.flow.network.authorisation import Priv
+from cylc.flow.parsec.config import ParsecConfig
+from cylc.flow.parsec.upgrade import upgrader
+from cylc.flow.parsec.validate import (
+    DurationFloat, CylcConfigValidator as VDR, cylc_config_validate)
+from cylc.flow.hostuserutil import get_user_home, is_remote_user
+
+# Nested dict of spec items.
+# Spec value is [value_type, default, allowed_2, allowed_3, ...]
+# where:
+# - value_type: value type (compulsory).
+# - default: the default value (optional).
+# - allowed_2, ...: the only other allowed values of this setting (optional).
+SPEC = {
+    'meta': {
+        'description': [VDR.V_STRING, ''],
+        'group': [VDR.V_STRING, ''],
+        'title': [VDR.V_STRING, ''],
+        'URL': [VDR.V_STRING, ''],
+        '__MANY__': [VDR.V_STRING, ''],
+    },
+    'general': {
```

Comment:
I think we should probably have a discussion and decide this - There are lots of possible names "cylc general", "cylc misc" , "suite settings", "suite global settings" or just "settings"
"suite general settings"
Personally I'm happy with more verbose & explicit rather than brief & fuzzy (🐁).

Reply:
moved to #3404